packages/components/src/components/types/icons.types.ts

```diff
@@ -0,0 +1,30 @@
+export interface IIconsManifest {
+    // If a new icon category is added, be sure to add it here as well. This is part of the fix for ts(7053)
+    readonly brand: string;
+    readonly cashier: string;
+    readonly common: string;
+    readonly contract: string;
+    readonly currency: string;
+    readonly dxtrade: string;
+    readonly flag: string;
+    readonly mt5: string;
+    readonly option: string;
+    readonly stock: string;
+    readonly tradetype: string;
+    readonly underlying: string;
+    readonly wallet: string;
+}
+export type TIconProps = {
+    className?: string;
+    color?: string;
+    custom_color?: string;
+    data_testid?: string;
+    height?: number | string;
+    icon: string;
```

Comment:
@jim-deriv Would be better to generate the type for `icon` from `packages/components/stories/icon/icons.js` to both get type-safety and auto-complete.

Reply:
@farzin-fs, I'll get to that folder. Right now, I'm converting this because it was cause PRs like [this](https://github.com/binary-com/deriv-app/pull/6385) one to faile

Reply:
so in this case if we want to use a new icon set it will throw an error and developer should go and update the icons types too?
if so please describe this in the icon.tsx component too. so everyone can understand it easily